core/client/components/character-name-text.js

```diff
@@ -7,28 +7,80 @@ const style = {
   strokeSize: 3,
 };
 
+const baselineStyle = {
+  ...style,
+  fontSize: 12,
+};
+
 const defaultTint = 0xFFFFFF;
+const baselineOffset = 20;
 
-class CharacterNameText extends Phaser.GameObjects.Text {
-  constructor(scene, text, tintName) {
-    super(scene, 0, 0, text);
+class CharacterNameText {
+  constructor(scene, name, baseline, tintName) {
+    this.scene = scene;
+    this.name = new Phaser.GameObjects.Text(scene, 0, 0, name);
 
-    this.setFontFamily(style.font)
+    this.name.setFontFamily(style.font)
       .setFontSize(style.fontSize)
       .setStroke(style.strokeColor, style.strokeSize)
       .setDepth(99999)
-      .setOrigin(0.5)
-      .setTintFromName(tintName);
+      .setOrigin(0.5);
+
+    scene.add.existing(this.name);
+    this.setBaseline(scene, baseline);
+    this.setTintFromName(tintName);
+  }
+
+  setBaseline(scene, text) {
```

Comment:
Why do you take `scene` as an argument rather than using `this.scene`, that's set in the constructor?

Reply:
Yes indeed, nice catch, it make no sense, I just missed it when I refactored my code.